### DISCUSSION TITLE
Case sensitive & insensitive

### DISCUSSION THREAD

**Discussion by artesburger** [dadoonet/fscrawler#1565] [Q&A] [answered]
Hi there,

I have a question about elasticsearch & fscrawler, because we are here to index files.
But elasticsearch are case sensitive by default.
Do you now how to configure an index, or a search , to a do search with case insensitive on an index create with fscrawler.

Best regards,

**Comment by dadoonet** [comment, marked as answer]
It depends. On which field do you want it to be case insensitive? Because on the `content` field, it's by default the case.

**Reply by artesburger** [reply]
Ok, it's good to know thanks you. But i want to do a search on filename. And the file name are often really complex like C-048D45.pdf.

**Reply by dadoonet** [reply]
I'd recommend [changing the mapping](https://fscrawler.readthedocs.io/en/latest/admin/fs/elasticsearch.html#mappings-1) in such a case. And change:

```json
        "file": {
          "properties": {
            "filename": {
              "type": "keyword",
              "store": true
            },
```

to:

```json
        "file": {
          "properties": {
            "filename": {
              "type": "text",
              "store": true
            },
```

That should work.

**Reply by artesburger** [reply]
Perfect !
